statsmodels/stats/robustros.py

```diff
@@ -0,0 +1,477 @@
+            else:
+                return row[self.result]
+
+        # create a DLIndex column that references self.cohn
+        self.data['DLIndex'] = self.data.apply(_ros_DL_index, axis=1)
+
+        # compute the ranks of the data
+        self.data['Rank'] = 1
+        groupcols = ['DLIndex', self.censorship, 'Rank']
+        rankgroups = (
+            self.data[groupcols]
+                .groupby(by=['DLIndex', self.censorship])
+        )
+        self.data['Rank'] = rankgroups.transform(lambda x: x.cumsum())
+
+        # detect/non-detect selectors
+        detect_selector = self.data[self.censorship] == False
+        nondet_selector = self.data[self.censorship] == True
+
+        # if there are no non-detects, just spit everything back out
+        if self.ncen == 0:
+            self.data['modeled'] = self.data[self.result]
+            self.data.sort(columns=[self.result], inplace=True)
+            ppos, sorted_res = stats.probplot(
+                self.data[self.result], fit=False
+            )
+            self.data['plot_pos'] = stats.norm.cdf(ppos)
+
+        # if there are too few detects, use half DL
+        elif self.nobs - self.ncen < 2 or self.ncen/self.nobs > 0.8:
+            self.data['modeled'] = self.data.apply(_select_half_DL, axis=1)
+            ppos, sorted_res = stats.probplot(
+                self.data[self.result], fit=False
+            )
+            self.data['plot_pos'] = stats.norm.cdf(ppos)
+
+        # in most cases, actually use the MR method to estimate NDs
+        else:
+            # compute the PE values
+            # (TODO: remove loop in place of apply)
+            for j in self.cohn.index[:-1][::-1]:
+                self.cohn.iloc[j]['prob_exceedance'] = (
+                    self.cohn.iloc[j+1]['prob_exceedance'] +
+                    self.cohn.iloc[j]['nuncen_above'] /
+                   (
+                        self.cohn.iloc[j]['nuncen_above'] +
+                        self.cohn.iloc[j]['nobs_below']
+                   ) * (1 - self.cohn.loc[j+1]['prob_exceedance'])
+                )
+
+            # compute the plotting position of the data (uses the PE stuff)
+            self.data['plot_pos'] = self.data.apply(_ros_plotting_pos, axis=1)
+
+            # correctly sort the plotting positions of the ND data:
+            ND_plotpos = self.data['plot_pos'][self.data[self.censorship]]
+            ND_plotpos.values.sort()
+            self.data.loc[self.data[self.censorship], 'plot_pos'] = ND_plotpos
+
+            # estimate a preliminary value of the Z-scores
+            self.data['Zprelim'] = stats.norm.ppf(self.data['plot_pos'])
+
+            # fit a line to the logs of the detected data
+            fit = stats.linregress(
+                self.data['Zprelim'][detect_selector],
+                self.transform_in(self.data[self.result][detect_selector])
+            )
+
+            # save the fit params to an attribute
+            self.fit = fit
```

Comment:
`self.fit_result` or something like this `self.results_fit` to avoid using attribute `fit`